owncloudData/src/main/java/com/owncloud/android/data/server/datasources/implementation/OCRemoteServerInfoDataSource.kt

```diff
@@ -41,9 +41,14 @@ class OCRemoteServerInfoDataSource(
         // Use the same client across the whole login process to keep cookies updated.
         val owncloudClient = clientManager.getClientForUnExistingAccount(path, false)
 
-        // Step 1: check whether the root folder exists.
-        val checkPathExistenceResult =
-            serverInfoService.checkPathExistence(path, isUserLogged = false, client = owncloudClient)
+        var checkPathExistenceResult =
+            serverInfoService.checkPathExistence(path, isUserLogged = false, owncloudClient)
+        var redirectionLocation = checkPathExistenceResult.redirectedLocation
+        while (!redirectionLocation.isNullOrEmpty()) {
+            checkPathExistenceResult =
+                serverInfoService.checkPathExistence(redirectionLocation, isUserLogged = false, owncloudClient)
+            redirectionLocation = checkPathExistenceResult.redirectedLocation
+        }
```

Comment:
Is this change required? We removed this loop some PR ago. We should use the same client that requested the remote status, so it should follow redirections by itself. That's why we removed it.

Is it needed now after removing the singleton?